Set the trove service password to random

s-t-e-v-e-n-k opened the pull request:
Force the trove service password to be random_password in the
model, to match other services.

rhafer commented:
I think we should also create a schema migration in https://github.com/crowbar/crowbar-openstack/tree/master/chef/data_bags/crowbar/migrate/trove that, if the current pw is set to ```trove``` resets it to a random password.

tboerger commented:
And the bot should exclude former team mates.

s-t-e-v-e-n-k commented:
@rhafer I will investigate how feasible that is.

dirkmueller commented on chef/data_bags/crowbar/template-trove.json:
```diff
@@ -25,7 +26,7 @@
-      "crowbar-revision": 1,
+      "crowbar-revision": 2,
       "crowbar-applied": false,
       "schema-revision": 100,
```
schema-revision needs to be bumped to 101

dirkmueller commented on chef/data_bags/crowbar/template-trove.json:
```diff
@@ -25,7 +26,7 @@
   "deployment": {
     "trove": {
-      "crowbar-revision": 1,
+      "crowbar-revision": 2,
```
that doesn't exist, it should stay with 1

dirkmueller commented on chef/data_bags/crowbar/migrate/trove/101_add_service_user.rb:
```diff
@@ -0,0 +1,9 @@
+
+def downgrade(ta, td, a, d)
+  a.delete("service_user")
+  return a, d
```
service_password needs to be deleted as well

dirkmueller commented on chef/data_bags/crowbar/migrate/trove/101_add_service_user.rb:
```diff
@@ -0,0 +1,9 @@
+def upgrade(ta, td, a, d)
+  a["service_user"] = ta["service_user"]
+  return a, d
```
service_password needs to be initialized to something sensible here

dirkmueller commented on chef/data_bags/crowbar/migrate/trove/101_add_service_user.rb:
```diff
@@ -0,0 +1,9 @@
+end
+
+def downgrade(ta, td, a, d)
+  a.delete("service_user")
```
do we need to back port this to cloud 6 as this smells like a security bug?

rhafer replied:
> do we need to back port this to cloud 6 as this smells like a security bug?

Yes, definitely.

rhafer commented:
Please add the bsc reference to the summary line of the commit message. That way our scripts will catch it for the changelog of the RPM packages.

rhafer commented:
BTW, for the generation of the password in the schema migration feel free to get some inspiration from: https://github.com/crowbar/crowbar-openstack/blob/master/chef/data_bags/crowbar/migrate/glance/004_add_db.rb

It's a little tricky since the migration is usually run two times on systems where trove is already deployed (1st for the Proposal, stored in crowbar's datbase, 2nd for the chef-role that holds the currently deployed configuration). The password needs to be set to the same value for both runs, that's why in class variable is needed there. (I think you can ignore the stuff dealing with node attribute from the above reference migration)

dirkmueller commented:
@s-t-e-v-e-n-k ping? :)

Itxaka commented on chef/data_bags/crowbar/template-trove.schema:
```diff
@@ -20,6 +20,8 @@
             "rabbitmq_instance": { "type": "str", "required": true },
             "volume_support": { "type": "bool", "required": true },
+            "service_user": { "type": "str", "required": true },
+            "service_password": { "type": "str" },
```
service_password in on the schema but not on the json. Shouldn't it be?

s-t-e-v-e-n-k replied:
We can't set it to a reasonable value in the JSON, since we want it to a random password.

Itxaka commented:
Apart from the small nit, looks good.

Itxaka commented:
retriggering jenkins as it seems that the job never posted back :/

nicolasbock commented:
+1

vuntz commented on chef/data_bags/crowbar/migrate/trove/101_add_service_details.rb:
```diff
@@ -0,0 +1,22 @@
+      node[:trove][:service_password] = @@trove_service_password
+      node.save
+    end
+  end
```
I don't think we need to change the nodes; the migration will change the role for the proposals, so the nodes will automatically get the attribute.

s-t-e-v-e-n-k commented:
Thanks for the review @vuntz. You're right that the node setting isn't required. I added the code since during my testing, I saw the updated password in the crowbar database, but not in the output of crowbar machine show. I've re-tested it without that chunk of code, and confirmed that it does show up in crowbar machine show after chef-client has run, so I've removed it.

vuntz commented:
Thanks, +1

Itxaka commented:
Looks good, thanks for addressing my comment +1

Itxaka commented:
retriggered the job :)
